docs/sections/components/llm/index.md

```diff
@@ -0,0 +1,305 @@
+        {"role": "system", "content": "You are Simpsons' fans who loves assigning grades from A to E, where A is the best and E is the worst."},
+        {"role": "user", "content": "What's up with Homer Simpson?"}
+    ]],
+    max_new_tokens=200
+)
+```
+
+We can check the output by parsing the content using the same pattern we required from the LLM.
+
+```python
+import re
+match = re.search(pattern, result[0][0])
+
+if match:
+    name = match.group(1)
+    grade = match.group(2)
+    print(f"Name: {name}, Grade: {grade}")
+# Name: Homer Simpson, Grade: C+
+```
+
+These were some simple examples, but one can see the options this opens.
+
+!!! NOTE
+    A complete pipeline can using [`LlamaCppLLM`][distilabel.llms.LlamaCppLLM] and *JSON* be seen in the [`examples/rpg_characters_llamacpp.py`](https://github.com/argilla-io/distilabel/tree/main/examples/rpg_characters_llamacpp.py) script.
+
+
+[^1]:
+    You can check the variable type by importing it from:
+
+    ```python
+    from distilabel.steps.tasks.structured_outputs.outlines import StructuredOutputType
+    ```
+
+[^2]:
+    Download the model with curl:
+
+    ```bash
+    curl -L -o ~/Downloads/openhermes-2.5-mistral-7b.Q4_K_M.gguf https://huggingface.co/TheBloke/OpenHermes-2.5-Mistral-7B-GGUF/resolve/main/openhermes-2.5-mistral-7b.Q4_K_M.gguf
+    ```
+
+#### OpenAI JSON
+
+OpenAI offers a [JSON Mode](https://platform.openai.com/docs/guides/text-generation/json-mode) to deal with structured output via their API, let's see how to make use of them. The JSON mode instructs the model to always return a JSON object following the instruction required.
+
+!!! WARNING
+    Bear in mind, that in order for this to work, you must instruct the model in some way to generate JSON, either in the `system message` or in the instruction, as can be seen in the [API reference](https://platform.openai.com/docs/guides/text-generation/json-mode).
+
+Contrary to what we have via `outlines`, JSON mode will not guarantee the output matches any specific schema, only that it is valid and parses without errors. More information can be found the OpenAI documentation.
+
+Other than the reference to generating JSON, to ensure the model generates parseable JSON we can pass the argument `response_format="json"`[^3]:
+
+```python
+from distilabel.llms import OpenAILLM
+llm = OpenAILLM(model="gpt4-turbo", api_key="api.key")
+llm.generate(..., response_format="json")
+```
+
+[^3]:
+    Keep in mind that to interact with this `response_format` argument in a pipeline, you will have to pass it via the `generation_kwargs`:
+
+    ```python
+    # Assuming a pipeline is already defined, and we have a task using OpenAILLM called `task_with_openai`:
+    pipeline.run(
+        parameters={
+            "task_with_openai": {
+                "llm": {
+                    "generation_kwargs": {
+                        "response_format": "json"
+                    }
+                }
+            }
+        }
+    )
+    ```
```

Comment:
And I believe this section may be a bit complex to include here, maybe we could offer this as an advanced thing under `docs/components/advanced/structured_generation.md` or something similar, to me that's more intuitive (plus adding the examples as already discussed)

Reply:
yes, will move it in a different PR